sootup.core/src/main/java/sootup/core/jimple/common/stmt/AbstractStmt.java

```diff
@@ -145,7 +145,12 @@ public StmtPositionInfo getPositionInfo() {
   @Nullable
   public Stmt withNewUse(@Nonnull Value oldUse, @Nonnull Value newUse) {
     ReplaceUseStmtVisitor visitor = new ReplaceUseStmtVisitor(oldUse, newUse);
-    accept(visitor);
+    try {
+      accept(visitor);
+    } catch (ClassCastException cce) {
+      // new Stmt is not created as the newUse could not be replaced
+      return null;
```

Comment:
how about optional instead of a possible null as return

Reply:
We could do that.. but would be structurally different to the other `with()`-ers